tpu/src/test/java/tpu/QueuedResourceIT.java

```diff
@@ -16,57 +16,89 @@
-import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Timeout;
 import org.junit.runner.RunWith;
 import org.junit.runners.JUnit4;
+import org.mockito.MockedStatic;
 
 @RunWith(JUnit4.class)
-@Timeout(value = 6, unit = TimeUnit.MINUTES)
+@Timeout(value = 3, unit = TimeUnit.MINUTES)
 public class QueuedResourceIT {
-  private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
+  private static final String PROJECT_ID = "project-id";
   private static final String ZONE = "europe-west4-a";
-  private static final String NODE_NAME = "test-tpu-queued-resource-network-" + UUID.randomUUID();
+  private static final String NODE_NAME = "test-tpu";
   private static final String TPU_TYPE = "v2-8";
   private static final String TPU_SOFTWARE_VERSION = "tpu-vm-tf-2.14.1";
-  private static final String QUEUED_RESOURCE_NAME = "queued-resource-network-" + UUID.randomUUID();
+  private static final String QUEUED_RESOURCE_NAME = "queued-resource";
   private static final String NETWORK_NAME = "default";
 
-  public static void requireEnvVar(String envVarName) {
-    assertWithMessage(String.format("Missing environment variable '%s' ", envVarName))
-        .that(System.getenv(envVarName)).isNotEmpty();
-  }
+  @Test
+  public void testCreateQueuedResourceWithSpecifiedNetwork() throws Exception {
+    TpuClient mockTpuClient = mock(TpuClient.class);
+    try (MockedStatic<TpuClient> mockedTpuClient = mockStatic(TpuClient.class)) {
+      mockedTpuClient.when(() -> TpuClient.create(any(TpuSettings.class)))
+          .thenReturn(mockTpuClient);
+
+      OperationFuture mockFuture = mock(OperationFuture.class);
+      when(mockTpuClient.createQueuedResourceAsync(any(CreateQueuedResourceRequest.class)))
+          .thenReturn(mockFuture);
+      CreateQueuedResourceWithNetwork.createQueuedResourceWithNetwork(
+          PROJECT_ID, ZONE, QUEUED_RESOURCE_NAME, NODE_NAME,
+          TPU_TYPE, TPU_SOFTWARE_VERSION, NETWORK_NAME);
 
-  @BeforeAll
-  public static void setUp() {
-    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
-    requireEnvVar("GOOGLE_CLOUD_PROJECT");
+      verify(mockTpuClient, times(1))
+          .createQueuedResourceAsync(any(CreateQueuedResourceRequest.class));
+    }
   }
 
-  @AfterAll
-  public static void cleanup() {
-    DeleteForceQueuedResource.deleteForceQueuedResource(PROJECT_ID, ZONE, QUEUED_RESOURCE_NAME);
+  @Test
+  public void testGetQueuedResource() throws IOException {
+    GetQueuedResource mockGetQueuedResource = mock(GetQueuedResource.class);
+    QueuedResource mockQueuedResource = mock(QueuedResource.class);
+    try (MockedStatic<TpuClient> mockedTpuClient = mockStatic(TpuClient.class)) {
+      mockedTpuClient.when(TpuClient::create).thenReturn(mock(TpuClient.class));
+      when(mock(TpuClient.class)
+          .getQueuedResource(any(QueuedResourceName.class))).thenReturn(mockQueuedResource);
+
+      GetQueuedResource.getQueuedResource(PROJECT_ID, ZONE, NODE_NAME);
```

Comment:
please add assert to verify that the return value is one that is expected to be returned by `getQueuedResource()` call or whatever the code sample method returns.

Reply:
Since we are mocking TpuClient, we can expect that the method has been called but no resources have been created before, so the return value of GetTpuVm.getTpuVm() will be null. Based on this example https://github.com/GoogleCloudPlatform/java-docs-samples/blob/f89f6faf150a64aacc49f9e4683c68a3c5000d7b/managedkafka/examples/src/test/java/examples/TopicsTest.java#L64 we can check that a String is printed when the method is running. Should I add a print like this to verify this? If we want to do E-2-E testing, we can move on to creating a real TpuClient and real resources.

Reply:
See my response in [another comment thread](https://github.com/GoogleCloudPlatform/java-docs-samples/pull/9679#discussion_r1842810183)

Reply:
Thank you! Added verification